DEVELOPMENT.md

```diff
@@ -0,0 +1,82 @@
+# Development Guide
+
+> **Note**
+> This assumes that you have already followed the steps in the [README.md](./README.md) to set-up your OpenShift Clusters, ODH, ACM, OSP, etc.
+
+## Useful Information
+
+### Models/Apps
+
+The GitHub repository is split into two main parts, Pipelines and ACM. Working on one part usually doesn’t involve working with the other, so you usually won’t have to set-up both dev environments.
+
+One aspect that is similar across both are the *models*/*apps*:
+
+* *Models* (used by Pipelines): These are the actual trained model files generated by Python ([`tf2model/`](https://www.tensorflow.org/guide/checkpoint), [`model.pkl`](https://scikit-learn.org/stable/model_persistence.html), etc.)
+* *Apps* (used by ACM): TL;DR: Sets of `YAML` files that reference images built using Pipelines and uploaded to Quay. See [RHACM Managing Applications](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/applications/managing-applications).
```

Comment:
I think "acm application" should be changed to "ACM Application", since we're talking about the ACM specific term "Application" which is different from the dictionary one. I also think the sentences should be switched for clarity e.g.
> "Each model is packaged in its [own container image](https://github.com/opendatahub-io/ai-edge/blob/e0086c9e2c223c8d4a1df7a415d8aee736365e75/pipelines/containerfiles/Containerfile.openvino.mlserver.mlflow) alongside a model server e.g. [OVMS](https://docs.openvino.ai/2023.1/ovms_what_is_openvino_model_server.html). These images are then built using the pipelines, pushed to Quay, and deployed as [ACM Applications](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/applications/managing-applications)"

Wdyt?

Reply:
> "Packaged into container image and these images then built by pipelines" ... that does not sound correct.

Now that you mention it I can see how that's confusing, how about "packaged in its own container image [...] These images are built using the pipelines"? Like how Ahmed wrote. "Model package" sounds wrong to me too for some reason but I think input from other people would be helpful on this.

Reply:
I don't ever remember us fetching from S3 _at runtime_. Fetching the model from S3 was in the Containerfile at one point though: https://gist.github.com/israel-hdez/1848d5a9788140c6f2e42ce316c70d5b#file-containerfile-with-storage-download-L1-L23